.azure-pipelines/build-job.yml

```diff
@@ -172,6 +172,12 @@ jobs:
         summaryFileLocation: _reports/**/Cobertura.xml
         pathToSources: src
 
+  - ${{ if eq(parameters.os, 'osx') }}:
+    - task: UseDotNet@2
```

Comment:
Is this relevant change for this PR? If so - could you please share more about it in description?

Reply:
That is not directly related to the migration, it is required to fix the issue with macOS release flow, in fact, it just porting changes from this PR - https://github.com/microsoft/azure-pipelines-agent/pull/3766 to fix macOS build.